### ISSUE TITLE
DKIM: Make d= tag configurable

### ISSUE THREAD

**Issue by caugner** [hmailserver/hmailserver#326]
I would like to use one DKIM selector for several subdomains:

```
myselector._domainkey.example.com IN TXT ...
```

However, I cannot use `myselector` for a subdomain `foo.example.com`, because hMailServer implicitly sets the domain tag to the current domain (`d=foo.example.com`). As a consequence, DKIM validation fails.

As a workaround, I could add the DKIM selector for each subdomain:

```
myselector._domainkey.foo.example.com IN TXT ...
myselector._domainkey.bar.example.com IN TXT ...
```

In practice, however, this is dangerous and cumbersome, because records for `example.com` are no longer implicitly valid for the subdomains (most importantly the `A`  and `MX` record, causing email and website to stop working) and I would have to always keep them synchronized.

**Comment by RvdHout** [comment]
> In practice, however, this is dangerous and cumbersome, because records for `example.com` are no longer implicitly valid for the subdomains (most importantly the `A` and `MX` record, causing email and website to stop working) and I would have to always keep them synchronized.

I don't get what you are saying here... i used exactly what you described in your workaround but i can not understand why this would break website and/or e-mail, can you please clarify?

You simply can have 1 MX-record for
example.com
a.example.com
b.example.com

All 3 can have a A-record, or one a-record en 2 cname records

**Comment by caugner** [comment]
```
example.com IN A 1.2.3.4
example.com IN MX 1.2.3.4
*.example.com IN A 1.2.3.4
```

In this example, DNS will return 1.2.3.4 as A and example.com as MX for all subdomains.

Let's add one record:

```
selector._domainkey.foo.example.com IN TXT "v=dkim;..."
```

Now DNS will return empty responses for A / MX for foo.example.com.

I guess this behavior could vary between DNS server software, but at least I experienced this problem.

PS: That's just the background of my feature request.

**Comment by RvdHout** [comment]
That seems wrong, the * in '*.example.com IN A 1.2.3.4' is a wildcard character, it should match everything

Try another DNS server software

**Comment by caugner** [comment]
This is the expected behavior. If you don't believe me, please read the corresponding RFC about wildcards in DNS: https://tools.ietf.org/html/rfc4592

tl;dr The wildcard only applies to non-existing subtrees.

Alternatively, just ignore the background of my feature request. 

**Feature Request**: I would like to customize the `d=` tag of the DKIM header. This allows me to specify one selector at the domain level and reuse it for any subdomain (even new subdomains, without requiring new DNS records).

**Comment by RvdHout** [comment]
What? 
In your example foo.example.com is non-existing, so it should fallback to *.example.com IN A 1.2.3.4

Do a DNS lookup for A-records for lets say: corona.d-fault.nl

```
; <<>> Dig.Net 1.0.0.1 <<>> @192.168.2.254 ANY corona.d-fault.nl
;; global options: printcmd
;; Got answer:
;; ->>HEADER<<- opcode: Query, status: NoError, id: 17105
;; flags:  qr rd ra; QUERY: 1, ANSWER: 5, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;corona.d-fault.nl.              	IN	ANY

;; ANSWER SECTION:
corona.d-fault.nl                3600	IN	MX	30 etrn.is.nl
corona.d-fault.nl                3600	IN	MX	10 mail.d-fault.nl
corona.d-fault.nl                297	IN	A	62.131.46.221
corona.d-fault.nl                297	IN	RRSIG	not-used
corona.d-fault.nl                3600	IN	RRSIG	not-used

;; Query time: 171 msec
;; SERVER: 192.168.2.254#53(192.168.2.254)
;; WHEN: Fri Mar 27 10:09:41 2020
;; MSG SIZE rcvd: 436
```

corona.d-fault.nl is non-existing, so it falls back to *.d-fault.nl which is a existing record

**Comment by caugner** [comment]
Please add the following record and see what happens:

```
example._domainkey.corona.d-fault.nl IN TXT "test"
```

**Comment by RvdHout** [comment]
```
; <<>> Dig.Net 1.0.0.1 <<>> @192.168.2.254 TXT example._domainkey.corona.d-fault.nl
;; global options: printcmd
;; Got answer:
;; ->>HEADER<<- opcode: Query, status: NoError, id: 11154
;; flags:  qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;example._domainkey.corona.d-fault.nl.	IN	TXT

;; ANSWER SECTION:
example._domainkey.corona.d-fault.nl 3600	IN	TXT	"test"

;; Query time: 115 msec
;; SERVER: 192.168.2.254#53(192.168.2.254)
;; WHEN: Fri Mar 27 11:55:22 2020
;; MSG SIZE rcvd: 71
```

**Comment by caugner** [comment]
And you're still getting the wildcard A value for the subdomain?

**Comment by RvdHout** [comment]
Hey, that's weird, now i see you point....
don't use a wildcard then, simply use a CNAME or A-record

**Comment by RvdHout** [comment]
corona.d-fault.nl now is a CNAME to d-fault and now it returns A-record again

**Comment by caugner** [comment]
That just doesn't work for me: I need wildcards, because I add subdomains frequently on the system running hMailServer and that system should not have any write access to the DNS server. 

Like I'm saying (hence this feature request), the real solution is to add the DKIM selector once for the domain and use it for the subdomains. This requires setting the DKIM d tag to that domain accordingly, which is currently not supported by hMailServer.

**Comment by RvdHout** [comment]
And I said, don't get you hopes up... at least anytime soon so if that doen't work for you hmailserver is not your mailserver software

**Comment by martinknafve** [comment]
This won't be implemented, due to lack of time on my part.
